PCbuild/_socket.vcxproj

```diff
@@ -60,6 +60,9 @@
     <_ProjectFileVersion>10.0.30319.1</_ProjectFileVersion>
   </PropertyGroup>
   <ItemDefinitionGroup>
+    <ClCompile>
+      <PreprocessorDefinitions>_WINSOCK_DEPRECATED_NO_WARNINGS;%(PreprocessorDefinitions)</PreprocessorDefinitions>
+    </ClCompile>
```

Comment:
This is the only change that I have a bit of a problem with: _socket is an internal module, so we should fix warnings rather than papering over them, if possible.  How much effort would it take to avoid the deprecated bits?